OpenRA.Game/Network/Connection.cs

```diff
@@ -325,6 +335,30 @@ void NetworkConnectionReceive()
 			}
 		}
 
+		void Ack(byte[] buf)
+		{
+			var reader = new BinaryReader(new MemoryStream(buf));
+			var frameReceived = reader.ReadInt32();
+			reader.ReadByte();
+			var framesToAck = reader.ReadInt16();
+
+			var ms = new MemoryStream();
```

Comment:
Avoid instantiating a empty memoryStream... the first write on it will secure 256 bytes of storage, which I think is not what you want...